### PR TITLE
Pull #13556: setup UnnecessaryOkComment module and suppressions (related to…

### PR DESCRIPTION
**Pull request by nrmancuso**
Related to https://github.com/checkstyle/checkstyle/issues/13213, this PR removes the old StyleValidationCommentInInputs module that demanded the `// ok` comments, and replaced it with this new one. I have generated all suppressions by file. In the scope of #13213, we can have contributors pick one file (or a few) and update the input files.

### CODE REVIEW

**Review by romani — APPROVED** [review]
ok to merge.

simple "//ok" should be forbidden, we need "// ok some wording to explain"

File: config/checkstyle-input-checks.xml [hunk 1]
```diff
@@ -24,12 +24,12 @@
 
   <module name="TreeWalker">
     <module name="Regexp">
-      <property name="id" value="StyleValidationCommentInInputs"/>
```
Comment (rnveach) [inline config/checkstyle-input-checks.xml:27]: https://github.com/search?q=repo%3Acheckstyle%2Fcheckstyle%20StyleValidationCommentInInputs&type=code

Old suppressions need to be removed.

File: config/checkstyle-input-checks.xml [hunk 2]
```diff
@@ -24,12 +24,12 @@
...
+      <property name="id" value="UnnecessaryOkComment"/>
       <property name="format"
-               value="(\/\/ (ok|violation|\d violations|filtered violation))"/>
-      <message key="required.regexp"
-             value="All check input files in the ''src/(it|test)/resources*/'' folder must
-                   have trailing comments &quot;// ok&quot; and/or &quot;// violation&quot;."/>
+               value="(?i)\/\/ ok$"/>
```
Comment (rnveach) [inline config/checkstyle-input-checks.xml:29]: What is `(?i)`?

Comment (rnveach) [inline config/checkstyle-input-checks.xml:29]: `\/\/ ok$` => `\/\/\s*ok$` ?

File: config/checkstyle-input-suppressions.xml [hunk 3]
```diff
@@ -202,4 +202,8255 @@
   <suppress id="StyleValidationCommentInInputs"
             files="regexp[\\/]regexponfilename[\\/]InputRegexpOnFilenameSemantic.java"/>
```
Comment (rnveach) [inline config/checkstyle-input-suppressions.xml:203]: See above.

**Review by rnveach — CHANGES_REQUESTED** [review]

File: config/checkstyle-input-checks.xml (hunk 2 above)
Comment (nrmancuso) [reply config/checkstyle-input-checks.xml:29]: Case insensitive mode flag.

Comment (nrmancuso) [reply config/checkstyle-input-checks.xml:29]: Found nine more violations, good idea.

File: config/checkstyle-input-checks.xml (hunk 1 above)
Comment (nrmancuso) [reply config/checkstyle-input-checks.xml:27]: Done:
```bash
➜  checkstyle git:(master) ag -Q "StyleValidationCommentInInputs" | wc -l
20


```

```bash
➜  checkstyle git:(issue-13213) ag -Q "StyleValidationCommentInInputs" | wc -l
0


```

File: config/checkstyle-input-checks.xml (hunk 2 above)
Comment (nrmancuso) [reply config/checkstyle-input-checks.xml]: Done

File: config/checkstyle-input-suppressions.xml (hunk 3 above)
Comment (nrmancuso) [reply config/checkstyle-input-suppressions.xml]: Done, see above

File: config/checkstyle-input-checks.xml (hunk 2 above)
Comment (rnveach) [reply config/checkstyle-input-checks.xml]: We should make a new issue for this.

`RegexpMultiline` and `RegexpSingleline` and `RegexpSinglelineJava` has `ignoreCase` but for some reason `RegExp` does not.

**Review by rnveach — APPROVED** [review]
@nrmancuso The commit should either start with `supplemental` or `issue`.

**Comment by romani** [comment]
Better to put PR number, as update endup as not easy and minor:)

File: config/checkstyle-input-checks.xml (hunk 2 above)
Comment (romani) [reply config/checkstyle-input-checks.xml]: I think we added such properties as we did not know it was possible to be done by regexp. It was big surprise for me that such mode exists.

Do we really need to add  such property?

**Comment by nrmancuso** [comment]
> Better to put PR number, as update endup as not easy and minor:)

Done

File: config/checkstyle-input-checks.xml (hunk 2 above)
Comment (nrmancuso) [reply config/checkstyle-input-checks.xml]: I do not think we such a property, but we should probably have some examples with this flag to demonstrate how to activate case insensitive mode.

**Pull request merged**